src/pcre2_compile.c

```diff
@@ -6479,6 +6485,9 @@ for (;; pptr++)
 #ifdef SUPPORT_WIDE_CHARS  /* Defined for 16/32 bits, or 8-bit with Unicode */
     if ((xclass_props & XCLASS_REQUIRED) != 0)
       {
+      /* We should never generate a (useless) xclass in 8-bit library when UTF flag is false */
+      PCRE2_ASSERT(PCRE2_CODE_UNIT_WIDTH != 8 || utf);
```

Comment:
This is usually an `#if` since it looks better, even if this variant is working.